src/browser/deck/actions.js

```diff
@@ -196,6 +196,20 @@ define((require, exports, module) => {
 
   const prepend = (items, item) => insert(items, item, 0);
 
+  const maybeSelectIndex = (items, index) => {
+    if (index >= items.length) {
+      return items;
+    }
+
+    const selectTab = advance(switchSelected,
```

Comment:
You could just do: `return activate(switchSelected(items, indexOfSelected(items), index));`